augur/filter.py

```diff
@@ -231,10 +233,26 @@ def run(args):
             num_excluded_by_length = len(seq_keep) - len(seq_keep_by_length)
             seq_keep = seq_keep_by_length
 
+    # filter by ambiguous dates
+    num_excluded_by_ambiguous_date = 0
+    if args.exclude_ambiguous_dates_by and 'date' in meta_columns:
+        seq_keep_by_date = []
+        for seq_name in seq_keep:
+            if is_date_ambiguous(meta_dict[seq_name]['date'],args.exclude_ambiguous_dates_by) is False:
+                seq_keep_by_date.append(seq_name)
+        num_excluded_by_ambiguous_date = len(seq_keep) - len(seq_keep_by_date)
+        seq_keep = seq_keep_by_date
+
     # filter by date
     num_excluded_by_date = 0
     if (args.min_date or args.max_date) and 'date' in meta_columns:
-        dates = get_numerical_dates(meta_dict, fmt="%Y-%m-%d")
+        if num_excluded_by_ambiguous_date:
+            date_meta_dict = {}
+            for seq_name in seq_keep:
+                date_meta_dict[seq_name]=meta_dict[seq_name]
+            dates = get_numerical_dates(date_meta_dict, fmt="%Y-%m-%d")
+        else:
+            dates = get_numerical_dates(meta_dict, fmt="%Y-%m-%d")
```

Comment:
meta_dict contains the entire meta_date, passing this to get_numerical_dates will throw the below error in case there is ambiguity in the month.
`    self.assert_only_less_significant_uncertainty()
  File "/augur/augur/util_support/date_disambiguator.py", line 122, in assert_only_less_significant_uncertainty
    "Invalid date: Month contains uncertainty, so day must also be uncertain."
ValueError: Invalid date: Month contains uncertainty, so day must also be uncertain.`

to avoid these kinds of errors, I got the dates only for the seq_keep.

Reply:
@saikirankv Can you say more about what data you're using when you get these errors? I don't see how the ambiguous date filter impacts the `get_numerical_dates` function and I don't get any errors when I revert this code back to its original state.